RISE-V2G-EVCC/src/main/java/com/v2gclarity/risev2g/evcc/transportLayer/UDPClient.java

```diff
@@ -106,6 +106,13 @@ public boolean initialize() {
 	
 
 	public static UDPClient getInstance() {
+		if (uniqueUDPClientInstance == null) {
+			synchronized (UDPClient.class) {
+				if (uniqueUDPClientInstance == null) {
+					uniqueUDPClientInstance = new UDPClient();
+				}
+			}
+		}
```

Comment:
In my opinion, there's no need to check if the uniqueUDPClientInstance is null as it is already set to new UDPClient() upon loading this class with the JVM (eager instantiation). Can you explain why you still see the need, or do you agree to my comment?

Reply:
I had to check this, because it can be null now: after a successful session I turn it to null, to get rid of the previous instance. So a new session will require a new instance.
I've reorganized the synchronization as well.